### DISCUSSION TITLE
Searching quote tweets?

### DISCUSSION THREAD

**Discussion by IBBoard** [CodedOre/NewCaw#34] [Backend]
Twitter have announced a QT search endpoint. It doesn't look like it will work for QTs in your notifications, but it would let us do "who QTed this?", similar to how we could do a "who RTed?" and "who liked?" display.

https://twittercommunity.com/t/introducing-the-quote-tweets-lookup-endpoint-to-the-twitter-api-v2/168370

Is this something that Mastadon would support? I think the old, conventional way of doing this is to search for the tweet URL.

**Comment by CodedOre** [comment]
Officially Mastodon don't has quotes, so they also don't have an api for getting who quoted a Post.
But they do have endpoints to get who has liked and reposted an post.

I don't think Cawbird currently does support this, but it would certainly be interesting to add.
Question is though how to implement it in the UI.

**Reply by IBBoard** [reply]
No, there's never been a "who RTed/liked", but I know other apps do it.

Based on the current Cawbird, I'd link to it from the "tweet detail" view (because it would be a lot of noise to have it in the main timeline). QTs might have to become a search because it would be a timeline-like set of tweets. RT/liked would need a different UI, potentially using the "search for people" widgets. I don't know how the list would fit with the new design, though.

**Reply by CodedOre** [reply]
Well, the current design looks like this:

![current_postview](https://user-images.githubusercontent.com/47981497/159119955-3af5d590-1908-4af7-859e-5e35397bb682.png)

I think we could implement it similar to how Twitter is doing it, by adding a row with the information on how many liked, reposted etc. and clicking on there would open a new view with the list:

![twitter_postview](https://user-images.githubusercontent.com/47981497/159120129-b383b2b9-db7c-4673-9bb3-9b73b6a18177.png)

**Reply by IBBoard** [reply]
Yeah, I think that works for individual tweets. I don't think tweet counts (even compact ones) are great in a timeline thought because a) it's lots of visual noise and b) it puts a lot of emphasis on engagement rather than content.
